code/game/shuttle_engines.dm

```diff
@@ -1,224 +1,597 @@
-	bound_height = 96
-	bound_width = 96
-	icon = 'icons/3x3.dmi'
-	icon_state = "huge_engine"
-	largeness = 2
+		src.updateUsrDialog()
+	if(href_list["admin_toggle_select_all"])
+		if(!isAdminGhost(usr))
+			to_chat(usr, "You must be an admin for this")
+			return
+
+		if(allow_selecting_all)
+			allow_selecting_all = 0
+			to_chat(usr, "Now selecting from shuttle's docking ports.")
+		else
+			allow_selecting_all = 1
+			to_chat(usr, "Now selecting from all existing docking ports.")
+
+		src.updateUsrDialog()
+	if(href_list["admin_reset"])
+		if(!isAdminGhost(usr))
+			to_chat(usr, "You must be an admin for this")
+			return
+
+		shuttle.initialize()
+		to_chat(usr, "Shuttle's list of travel destinations has been reset")
+	if(href_list["admin_toggle_silicon_use"])
+		if(!isAdminGhost(usr))
+			to_chat(usr, "You must be an admin for this")
+			return
+
+		if(allow_silicons)
+			allow_silicons = 0
+			to_chat(usr, "Silicons can no longer use [src].")
+		else
+			allow_silicons = 1
+			to_chat(usr, "Silicons may now use [src] again.")
+
+		src.updateUsrDialog()
+	if(href_list["disk"])
+		if(!disk) //No disk inserted - grab one from user's hand
+			var/obj/item/weapon/disk/shuttle_coords/D = usr.get_active_hand()
+
+			insert_disk(D, usr)
+		else
+			disk.forceMove(get_turf(src))
+			usr.put_in_hands(disk)
+			to_chat(usr, "<span class='info'>You eject \the [disk] from \the [src].</span>")
+			if(disk.destination == selected_port)
+				selected_port = null
+			disk = null
+			src.updateUsrDialog()
+
+/obj/machinery/computer/shuttle_control/proc/insert_disk(obj/item/weapon/disk/shuttle_coords/SC, mob/user)
+	if(!shuttle)
+		to_chat(user, "<span class='info'>\The [src] is unresponsive.</span>")
+		return
+
+	if(!istype(SC))
+		if(istype(SC, /obj/item/weapon/disk)) //It's a disk, but not a compactible one
+			to_chat(user, "<span class='info'>The disk is rejected by \the [src].</span>")
+
+		return
+
+	if(disk)
+		//An old disk is already inserted.
+		to_chat(user, "<span class='warning'>The old [disk.name] pops out of the disk slot!</span>")
+		disk.forceMove(src.loc)
+		disk = null
+
+	if(user.drop_item(SC, src))
+		disk = SC
+		to_chat(user, "<span class='info'>You insert \the [SC] into \the [src].</span>")
+		src.updateUsrDialog()
+
+/obj/machinery/computer/shuttle_control/proc/use_pass(obj/item/weapon/card/shuttle_pass/P, mob/user)
+	if(!istype(P))
+		return
+
+	if(user.drop_item(P, src))
+		if(shuttle && shuttle.type == P.allowed_shuttle)
+			if(shuttle.travel_to(P.destination, src, user))
+				to_chat(user, "<span class='info'>You insert \the [P] into \the [src].</span>")
+				qdel(P)
+				return
+		to_chat(user, "<span class='info'>You insert \the [P] into \the [src], but it is rejected.</span>")
+		user.put_in_hands(P)
+
+/obj/machinery/computer/shuttle_control/bullet_act(var/obj/item/projectile/Proj)
+	visible_message("[Proj] ricochets off [src]!")
+	return ..() // Nothing happens (?)
```

Comment:
Yes, this would return `PROJECTILE_COLLISION_DEFAULT` which would absorb the hit and do nothing.